### PR TITLE
Update networking docs

### PR DESCRIPTION
**Pull request by mit-mit**
Fixes https://github.com/flutter/flutter/issues/13602

### CODE REVIEW

File: networking.md [hunk 1]
```diff
@@ -168,16 +176,13 @@ class _MyHomePageState extends State<MyHomePage> {
...
-[http]:       https://pub.dartlang.org/packages/http
-[client]:     https://www.dartdocs.org/documentation/http/latest/http/Client-class.html
-[get]:        https://www.dartdocs.org/documentation/http/latest/http/get.html
-[read]:       https://www.dartdocs.org/documentation/http/latest/http/read.html
-[readbytes]:  https://www.dartdocs.org/documentation/http/latest/http/readBytes.html
-[post]:       https://www.dartdocs.org/documentation/http/latest/http/post.html
-[convert]:    https://docs.flutter.io/flutter/dart-convert/dart-convert-library.html
 [dartio]:     https://api.dartlang.org/stable/dart-io/dart-io-library.html
+[convert]:    https://docs.flutter.io/flutter/dart-convert/dart-convert-library.html
+[client]:     https://api.dartlang.org/stable/dart-io/HttpClient-class.html
```
Comment (sethladd) [inline networking.md]: Can we link to the Flutter API docs for these? e.g. https://docs.flutter.io/flutter/dart-io/HttpClient-class.html

File: networking.md [hunk 2]
```diff
@@ -168,16 +176,13 @@ class _MyHomePageState extends State<MyHomePage> {
...
+  * [library `dart:convert`][convert]
 
-[http]:       https://pub.dartlang.org/packages/http
-[client]:     https://www.dartdocs.org/documentation/http/latest/http/Client-class.html
-[get]:        https://www.dartdocs.org/documentation/http/latest/http/get.html
-[read]:       https://www.dartdocs.org/documentation/http/latest/http/read.html
-[readbytes]:  https://www.dartdocs.org/documentation/http/latest/http/readBytes.html
-[post]:       https://www.dartdocs.org/documentation/http/latest/http/post.html
-[convert]:    https://docs.flutter.io/flutter/dart-convert/dart-convert-library.html
 [dartio]:     https://api.dartlang.org/stable/dart-io/dart-io-library.html
```
Comment (sethladd) [inline networking.md]: For all API docs, let's link to docs.flutter.io/something

File: networking.md [hunk 3]
```diff
@@ -125,18 +120,31 @@ class _MyHomePageState extends State<MyHomePage> {
 
   _getIPAddress() async {
     String url = 'https://httpbin.org/ip';
```
Comment (sethladd) [inline networking.md]: we could use var here.

File: networking.md [hunk 4]
```diff
@@ -10,54 +10,51 @@ sockets, see [dart:io][dartio].
...
-```
-dependencies:
-  ...
-  http: '>=0.11.3+12'
-```
-
 ## Making HTTP requests
 
-Next, create an HTTP [Client][client]. We recommend using
-[`createHttpClient`](https://docs.flutter.io/flutter/services/createHttpClient.html)
```
Comment (sethladd) [inline networking.md]: should we add a section to this doc for mocking/testing with HTTP?

File: networking.md (hunk 2 above)
Comment (mit-mit) [reply networking.md]: Oh, yes of course!

File: networking.md (hunk 4 above)
Comment (mit-mit) [reply networking.md]: We should, and I started doing it yesterday, but looks like the overrides APIs are still in review, so I'd rather do it in a second PR. Filed https://github.com/flutter/flutter/issues/13792 for that.

**Comment by mit-mit** [comment]
@sethladd thanks for feedback, PR updated, please take another look

**Comment by Hixie** [comment]
@mit-mit Is this ready to land?

**Comment by mit-mit** [comment]
Yup, merging!

**Pull request merged**